Build failure with Boost 1.81

## Describe the bug

Icinga2 2.13.6 fails to compile against Boost 1.81.

## To Reproduce

Provide a link to a live example, or an unambiguous set of steps to reproduce this bug. Include configuration, logs, etc. to reproduce, if relevant.

1. Upgrade Boost to Boost 1.81
2. Build Icinga with the following options:
`-DICINGA2_UNITY_BUILD=no -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DICINGA2_SYSCONFIGFILE=/etc/conf.d/icinga2 -DICINGA2_PLUGINDIR=/usr/lib64/nagios/plugins -DICINGA2_USER=icinga -DICINGA2_GROUP=icingacmd -DICINGA2_COMMAND_GROUP=icingacmd -DICINGA2_RUNDIR=/run -DINSTALL_SYSTEMD_SERVICE_AND_INITSCRIPT=ON -DUSE_SYSTEMD=yes -DLOGROTATE_HAS_SU=ON -DICINGA2_LTO_BUILD=yes -DICINGA2_WITH_PGSQL=no -DICINGA2_WITH_MYSQL=yes`
3. Build fails:
```
FAILED: lib/remote/CMakeFiles/remote.dir/httphandler.cpp.o 
/var/tmp/portage/net-analyzer/icinga2-2.13.6/temp/clang-wrappers/x86_64-pc-linux-gnu-g++ -DBOOST_ASIO_USE_TS_EXECUTOR_AS_DEFAULT -DBOOST_COROUTINES_NO_DEPRECATION_WARNING -DBOOST_FILESYSTEM_NO_DEPRECATED -D_GNU_SOURCE -I/var/tmp/portage/net-analyzer/icinga2-2.13.6/work/icinga2-2.13.6/third-party/nlohmann_json -I/var/tmp/portage/net-analyzer/icinga2-2.13.6/work/icinga2-2.13.6/third-party/utf8cpp/source -I/var/tmp/portage/net-analyzer/icinga2-2.13.6/work/icinga2-2.13.6 -I/var/tmp/portage/net-analyzer/icinga2-2.13.6/work/icinga2-2.13.6/lib -I/var/tmp/portage/net-analyzer/icinga2-2.13.6/work/icinga2-2.13.6_build -I/var/tmp/portage/net-analyzer/icinga2-2.13.6/work/icinga2-2.13.6_build/lib  -O2 -pipe -march=native -fdiagnostics-color=always -frecord-gcc-switches -Wreturn-type -D_GLIBCXX_ASSERTIONS -g -pthread -Winvalid-pch -flto -std=c++0x -fdiagnostics-color=always -MD -MT lib/remote/CMakeFiles/remote.dir/httphandler.cpp.o -MF lib/remote/CMakeFiles/remote.dir/httphandler.cpp.o.d -o lib/remote/CMakeFiles/remote.dir/httphandler.cpp.o -c /var/tmp/portage/net-analyzer/icinga2-2.13.6/work/icinga2-2.13.6/lib/remote/httphandler.cpp
/var/tmp/portage/net-analyzer/icinga2-2.13.6/work/icinga2-2.13.6/lib/remote/httphandler.cpp: In static member function ‘static void icinga::HttpHandler::ProcessRequest(icinga::AsioTlsStream&, const icinga::ApiUser::Ptr&, boost::beast::http::request<boost::beast::http::basic_string_body<char> >&, boost::beast::http::response<boost::beast::http::basic_string_body<char> >&, boost::asio::yield_context&, icinga::HttpServerConnection&)’:
/var/tmp/portage/net-analyzer/icinga2-2.13.6/work/icinga2-2.13.6/lib/remote/httphandler.cpp:61:49: error: ‘using string_view = boost::core::string_view’ {aka ‘class boost::core::basic_string_view<char>’} has no member named ‘to_string’
   61 |         Url::Ptr url = new Url(request.target().to_string());
      |                                                 ^~~~~~~~~
```

Full log: https://bugs.gentoo.org/attachment.cgi?id=844789.

## Expected behavior

Successful compilation.

## Your Environment

* Version used (`icinga2 --version`):
* Operating System and version: Gentoo, rolling distro
* Enabled features (`icinga2 feature list`): N/A
* Icinga Web 2 version and modules (System - About): N/A
* Config validation (`icinga2 daemon -C`): N/A
* If you run multiple Icinga 2 instances, the `zones.conf` file (or `icinga2 object list --type Endpoint` and `icinga2 object list --type Zone`) from all affected nodes: N/A

## Additional context

- This is a forward of a downstream report in Gentoo at https://bugs.gentoo.org/888063
- Alpine Linux has a patch at https://git.alpinelinux.org/aports/commit/?id=2255728eeb75fe2e71d58f4c9a00a661de5b7406

--- a/lib/base/string.cpp
+++ b/lib/base/string.cpp
@@ -128,15 +128,15 @@ String::operator const std::string&() const
 }
 
 /**
- * Conversion function to boost::string_view.
+ * Conversion function to boost::beast::string_view.
  *
  * This allows using String as the value for HTTP headers in boost::beast::http::basic_fields::set.
  *
- * @return A boost::string_view representing this string.
+ * @return A boost::beast::string_view representing this string.
  */
-String::operator boost::string_view() const
+String::operator boost::beast::string_view() const
 {
-	return boost::string_view(m_Data);
+	return boost::beast::string_view(m_Data);
 }
 
 const char *String::CStr() const

--- a/lib/base/string.hpp
+++ b/lib/base/string.hpp
@@ -5,6 +5,7 @@
 
 #include "base/i2-base.hpp"
 #include "base/object.hpp"
+#include <boost/beast/core.hpp>
 #include <boost/range/iterator.hpp>
 #include <boost/utility/string_view.hpp>
 #include <functional>
@@ -73,7 +74,7 @@ public:
 	bool operator<(const String& rhs) const;
 
 	operator const std::string&() const;
-	operator boost::string_view() const;
+	operator boost::beast::string_view() const;
 
 	const char *CStr() const;
 

--- a/lib/remote/httphandler.cpp
+++ b/lib/remote/httphandler.cpp
@@ -58,7 +58,7 @@ void HttpHandler::ProcessRequest(
 	Dictionary::Ptr node = m_UrlTree;
 	std::vector<HttpHandler::Ptr> handlers;
 
-	Url::Ptr url = new Url(request.target().to_string());
+	Url::Ptr url = new Url(std::string(request.target()));
 	auto& path (url->GetPath());
 
 	for (std::vector<String>::size_type i = 0; i <= path.size(); i++) {

--- a/lib/remote/httpserverconnection.cpp
+++ b/lib/remote/httpserverconnection.cpp
@@ -246,7 +246,7 @@ bool HandleAccessControl(
 			if (!allowedOrigins.empty()) {
 				auto& origin (request[http::field::origin]);
 
-				if (allowedOrigins.find(origin.to_string()) != allowedOrigins.end()) {
+				if (allowedOrigins.find(std::string(origin)) != allowedOrigins.end()) {
 					response.set(http::field::access_control_allow_origin, origin);
 				}
 
@@ -536,7 +536,7 @@ void HttpServerConnection::ProcessMessages(boost::asio::yield_context yc)
 			if (!authenticatedUser) {
 				CpuBoundWork fetchingAuthenticatedUser (yc);
 
-				authenticatedUser = ApiUser::GetByAuthHeader(request[http::field::authorization].to_string());
+				authenticatedUser = ApiUser::GetByAuthHeader(std::string(request[http::field::authorization]));
 			}
 
 			Log logMsg (LogInformation, "HttpServerConnection");

